pt_miniscreen/core/component.py

```diff
@@ -0,0 +1,223 @@
+import logging
+import threading
+from time import sleep, time
+from typing import Any, Dict
+from weakref import WeakMethod
+
+from PIL import Image
+
+from .utils import is_same_image
+
+logger = logging.getLogger(__name__)
+
+
+class Interval(threading.Timer):
+    def __init__(
+        self,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.daemon = True
+
+    @property
+    def function(self):
+        return self._get_function()
+
+    @function.setter
+    def function(self, next_value):
+        self._get_function = WeakMethod(next_value)
+
+    def run(self):
+        execution_time = 0
+
+        while not self.finished.is_set():
```

Comment:
I can't find reference to this `finished` event in the threading docs... am I missing something?